app/controllers/admin/custom_fields/hierarchy/items_controller.rb

```diff
@@ -184,6 +199,24 @@ def add_errors_to_edit_form(validation_result)
           end
         end
 
+        def parse_parent_input(new_parent_input)
+          case new_parent_input
+          in [new_parent]
+            input = MultiJson.load(new_parent, symbolize_keys: true)[:value]
+            new_parent = CustomField::Hierarchy::Item.including_children.find(input)
```

Comment:
:yellow_circle: Should we explicitly handle the case of not finding the item with the given value?

Right now this case would raise a `RecordNotFound` error, when I would expect a `Failure` to be returned.

Current behaviour:

<img width="822" height="141" alt="image" src="https://github.com/user-attachments/assets/f07fa390-7b3f-41ab-a86e-860077869e97" />

Reply:
How can you trigger it? But in general, you are right. handling the error more gracefully can only help

Reply:
Easy few steps :sunglasses: 

1. Open the assign new parent dialog and decide for a parent (no saving yet)
2. Open a new tab, delete the parent you decided for
3. Go back to the original tab and assign the parent

Reply:
true, very easy :D